lib/viewcomponents/io.jsx

```diff
@@ -0,0 +1,78 @@
+var React = require('react');
+var mouseDownDrag = require('../utilities/mouseDownDrag');
+var LabelPosition = require('../enum/ioLabelPosition');
+
+var ioRadius = 4;
+
+var IO = React.createClass({
+  propTypes: {
+    isInput: React.PropTypes.bool.isRequired,
+
+    label: React.PropTypes.string,
+
+    x: React.PropTypes.number.isRequired,
+
+    y: React.PropTypes.number.isRequired,
+
+    size: React.PropTypes.number,
+
+    labelPosition: React.PropTypes.oneOf(Object.keys(LabelPosition).map(function(k) {
+      return LabelPosition[k];
+    }))
```

Comment:
I wish there was a object.values (maybe I oughta make one). I'm trying to
see that it's a value in the enum.

On Sat, Feb 14, 2015, 12:20 AM Robert Gilmore <notifications@github.com>
wrote:

> In lib/viewcomponents/io.jsx
> <https://github.com/twosigma/goll-e/pull/5#discussion_r24712399>:
>
> > +
> > +var IO = React.createClass({
> > +  propTypes: {
> > +    isInput: React.PropTypes.bool.isRequired,
> > +
> > +    label: React.PropTypes.string,
> > +
> > +    x: React.PropTypes.number.isRequired,
> > +
> > +    y: React.PropTypes.number.isRequired,
> > +
> > +    size: React.PropTypes.number,
> > +
> > +    labelPosition: React.PropTypes.oneOf(Object.keys(LabelPosition).map(function(k) {
> > +      return LabelPosition[k];
> > +    }))
>
> I don't really understand what you're doing here. I get the oneOf and the
> map function, but what's up with Object.keys? I've never seen that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twosigma/goll-e/pull/5/files#r24712399>.
>